### PR TITLE
Add localized and localized(comment:) to String

### PR DESCRIPTION
**Pull request by Vyax**
## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [ ] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.

I wasn't able to write tests for these extensions, because they require running with a different localization. As far as I know, it is currently not possible to set the 'Application Language' for a single test.

### CODE REVIEW

File: Sources/Extensions/SwiftStdlib/StringExtensions.swift [hunk 1]
```diff
@@ -231,6 +231,13 @@ public extension String {
 	public var length: Int {
 		return characters.count
 	}
+    /// SwifterSwift: Returns a localized string.
+    ///
+    ///        "Hello!".localized -> Hallo!
+    ///
+    public var localized: String {
+        return NSLocalizedString(self, comment: "")
+    }
```
Comment (omaralbeik) [inline Sources/Extensions/SwiftStdlib/StringExtensions.swift]: I prefer having a single function here with a default value for comment like below:
```swift
func localized(comment: String = "") {
    ...
}
```

File: Sources/Extensions/SwiftStdlib/ArrayExtensions.swift [hunk 2]
```diff
@@ -56,6 +56,17 @@ public extension Array {
...
+    /// SwifterSwift: Get the n last elements.
+    ///
+    ///        [1, 7, 1, 2, 4, 1, 6].last(n: 3) -> [4, 1, 6]
+    ///
+    /// - Parameter n: maximum number of elements.
+    /// - Returns: array slice of the n last elements, otherwise an empty array
+    public func last(n: Int) -> ArraySlice<Element> {
+        guard n > 0 && !isEmpty else { return [] }
+        return self[Swift.max(endIndex - n, 0)...]
+    }
```
Comment (omaralbeik) [inline Sources/Extensions/SwiftStdlib/ArrayExtensions.swift]: @Vyax I noticed you have another PR #270 with the same function, please make sure it is available in one of them only :)

File: Sources/Extensions/SwiftStdlib/StringExtensions.swift (hunk 1 above)
Comment (Vyax) [reply Sources/Extensions/SwiftStdlib/StringExtensions.swift]: Thank you for the suggestion. However, I frequently use localized strings without comments and would prefer "HelloWorld".localized to "HelloWorld".localized().

File: Sources/Extensions/SwiftStdlib/ArrayExtensions.swift (hunk 2 above)
Comment (Vyax) [reply Sources/Extensions/SwiftStdlib/ArrayExtensions.swift]: Thank you, I removed the extraneous commit.

**Comment by Vyax** [comment]
Sorry for closing and reopening the pull request. Back to our conversation:
I frequently use localized strings without comments and would prefer "HelloWorld".localized to "HelloWorld".localized(). What do you think about it?

**Comment by omaralbeik** [comment]
@Vyax I'd still prefer to have a single function :)
Lets ask other members here: @SD10, @LucianoPAlmeida @pvtmert what do you think guys?

**Comment by LucianoPAlmeida** [comment]
Hey guys :) @omaralbeik, @Vyax , I like the localized to, but in my opinion I prefer a method with comment, because it's very useful when you work on teams to the other guys know what that localized key means. Maybe if you like use without comment you can define a default value like: 
``` swift 
      func localized(comment: String = "")  { 
            return NSLocalizedString(self, comment: comment)
      }
```
This way you can use both with or without comments :)

**Comment by Vyax** [comment]
Okay, 2 vs 1. I changed it to a single function with an optional parameter.

**Pull request merged**